tests/components/octoprint/test_config_flow.py

```diff
@@ -0,0 +1,331 @@
+    with patch(
+        "pyoctoprintapi.OctoprintClient.get_server_info",
+        side_effect=CannotConnect,
+    ), patch("pyoctoprintapi.OctoprintClient.request_app_key", return_value="test-key"):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "host": "1.1.1.1",
+                "username": "testuser",
+                "name": "Printer",
+                "port": 81,
+                "ssl": True,
+                "path": "/",
+            },
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_unknown_exception(hass):
+    """Test we handle a random error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "pyoctoprintapi.OctoprintClient.get_server_info",
+        side_effect=Exception,
+    ), patch("pyoctoprintapi.OctoprintClient.request_app_key", return_value="test-key"):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "host": "1.1.1.1",
+                "username": "testuser",
+                "name": "Printer",
+                "port": 81,
+                "ssl": True,
+                "path": "/",
+            },
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "unknown"}
+
+
+async def test_show_zerconf_form(hass: HomeAssistant) -> None:
+    """Test that the zeroconf confirmation form is served."""
+
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_ZEROCONF},
+        data={
+            "host": "192.168.1.123",
+            "port": 80,
+            "hostname": "example.local.",
+            "uuid": "83747482",
+            "properties": {"uuid": "83747482", "path": "/foo/"},
+        },
+    )
+    assert result["type"] == "form"
+    assert not result["errors"]
+
+    with patch(
+        "pyoctoprintapi.OctoprintClient.get_server_info",
+        return_value=True,
+    ), patch(
+        "pyoctoprintapi.OctoprintClient.get_discovery_info",
+        return_value=DiscoverySettings({"upnpUuid": "uuid"}),
+    ), patch(
+        "pyoctoprintapi.OctoprintClient.get_printer_info",
+        return_value=OctoprintPrinterInfo(
+            {
+                "state": {
+                    "flags": {"printing": True, "error": False},
+                    "text": "Operational",
+                },
+                "temperature": [],
+            }
+        ),
+    ), patch(
+        "pyoctoprintapi.OctoprintClient.get_job_info",
+        return_value=OctoprintJobInfo(
+            {
+                "job": {},
+                "progress": {"completion": 50},
+            }
+        ),
+    ), patch(
```

Comment:
We want to patch the integration setup functions on all create entry results to avoid setting up the whole integration.

Reply:
Done